src/math/sqrt_int16.c

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/sqrt.h>
+
+/* Defines Constant*/
+#define SQRT_WRAP_SCHAR_BITS 0xFF
+
+/* Function Definitions */
+/*
+ *  Square root
+ *
+ *  Y = SQRTLOOKUP_INT16(U) computes the square root of
+ *  U using lookup tables.
+ * +------------------+-----------------+--------+--------+
+ * | u		      | y (returntype)	|   u	 |  y	  |
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * | 16 | 12  |	 0    | 16  | 15 |   1	| 4.12	 | 2.14	  |
+ * +------------------+-----------------+--------+--------+
+
```

Comment:
I think the table should be like this to decsribe the function behavior based on my test run with TDFB component.

```
 * +------------------+-----------------+--------+--------+
 * | u		      | y (returntype)	|   u	 |  y	  |
 * +----+-----+-------+----+----+-------+--------+--------+
 * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
 * +----+-----+-------+----+----+-------+--------+--------+
 * | 16 | 12  |	 0    | 16  | 12 |   1	| 4.12	 | 4.12	  |
 * +------------------+-----------------+--------+--------+
```
You could also mention in comments that range of u is 0 - 16 and range of is y 0 - 4 to help understand the fractional format.

(The lib code works OK!)

Reply:
Sure